igvm/vm.py

```diff
@@ -1005,19 +1006,33 @@ def aws_get_instances_overview(
                 etag = None
 
             if file.exists() and etag_file.exists() and etag:
-                with open(AWS_INSTANCES_OVERVIEW_FILE_etag) as f:
+                with open(AWS_INSTANCES_OVERVIEW_FILE_etag, 'r+') as f:
                     prev_etag = f.read()
                 if etag == prev_etag:
-                    with open(AWS_INSTANCES_OVERVIEW_FILE) as f:
+                    with open(AWS_INSTANCES_OVERVIEW_FILE, 'r+') as f:
                         return json.load(f)
 
             resp = urlopen(url, timeout=timeout)
             if etag:
-                with open(AWS_INSTANCES_OVERVIEW_FILE_etag, 'w') as f:
+                with open(AWS_INSTANCES_OVERVIEW_FILE_etag, 'w+') as f:
                     f.write(etag)
-            with open(AWS_INSTANCES_OVERVIEW_FILE, 'w') as f:
+                    os.chmod(AWS_INSTANCES_OVERVIEW_FILE_etag,
+                             stat.S_IREAD |
+                             stat.S_IWRITE |
+                             stat.S_IRGRP |
+                             stat.S_IWGRP |
+                             stat.S_IROTH |
+                             stat.S_IWOTH)
```

Comment:
Why do we need to make the file world read and also world writable - can't we avoid that ?

Reply:
We can. I've added the file to `adm` group and removed the world writable flag. Reading does no harm